spec/pagseguro/authorization_spec.rb

```diff
@@ -48,39 +48,50 @@
     end
   end
 
-  describe "find_by_date" do
-    let(:date_options) { { initial_date: initial_date, final_date: final_date } }
-    let(:initial_date) { Time.new(2015, 11, 1, 12, 0) }
-    let(:final_date) { Time.new(2015, 11, 2, 12, 0) }
+  describe "find_by" do
     before do
-      expect(PagSeguro::Request).to receive(:get)
-        .with("authorizations", 'v2', { initialDate: initial_date.xmlschema, finalDate: final_date.xmlschema })
-        .and_return(request)
-      expect(PagSeguro::Authorization::Collection).to receive(:new)
+      allow(PagSeguro::Authorization::Collection).to receive(:new)
         .and_return(collection)
-      expect(PagSeguro::Authorization::Response).to receive(:new)
+      allow(PagSeguro::Authorization::Response).to receive(:new)
         .with(request, collection)
         .and_return(response)
-      expect(response).to receive(:serialize_collection)
+      allow(response).to receive(:serialize_collection)
     end
 
-    it 'finds authorizations by a date range' do
-      expect(PagSeguro::Authorization.find_by_date(date_options)).to eq (collection)
+    context "with reference" do
+      let(:reference) { 'REF1111' }
+
```

Comment:
I think is good to make a test when only credentials is passed.
This already solve the question `Consulta de autorização por aplicação`. :wink: